plugins/modules/system/ssh_config.py

```diff
@@ -0,0 +1,315 @@
+    choices: [ 'yes', 'no', 'ask' ]
+    type: str
+  proxycommand:
+    description:
+      - Sets the C(ProxyCommand) option.
+    type: str
+  ssh_config_file:
+    description:
+      - SSH config file.
+      - If I(user) is none, C(/etc/ssh/ssh_config) is used.
+      - Mutually exclusive with I(user).
+    type: path
+requirements:
+- StormSSH
+notes:
+- Support check mode.
+'''
+
+EXAMPLES = r'''
+- name: Add a host in the configuration
+  community.general.ssh_config:
+    user: akasurde
+    host: "example.com"
+    hostname: "github.com"
+    identity_file: "/home/akasurde/.ssh/id_rsa"
+    port: '2223'
+    state: present
+
+- name: Delete a host from the configuration
+  community.general.ssh_config:
+    ssh_config_file: "{{ ssh_config_test }}"
+    host: "example.com"
+    state: absent
+'''
+
+RETURN = r'''
+hosts_added:
+  description: A list of host added.
+  returned: success
+  type: list
+  sample: ["example.com"]
+hosts_removed:
+  description: A list of host removed.
+  returned: success
+  type: list
+  sample: ["example.com"]
+hosts_changed:
+  description: A list of host changed.
+  returned: success
+  type: list
+  sample: ["example.com"]
+hosts_change_diff:
+  description: A list of host diff changes.
+  returned: on change
+  type: list
+  sample: [
+    {
+      "example.com": {
+        "new": {
+          "hostname": "github.com",
+          "identityfile": ["/tmp/test_ssh_config/fake_id_rsa"],
+          "port": "2224"
+        },
+        "old": {
+          "hostname": "github.com",
+          "identityfile": ["/tmp/test_ssh_config/fake_id_rsa"],
+          "port": "2224"
+        }
+      }
+    }
+  ]
+'''
+
+import os
+import traceback
+
+from copy import deepcopy
+
+STORM_IMP_ERR = None
+try:
+    from storm.parsers.ssh_config_parser import ConfigParser
+    HAS_STORM = True
+except ImportError:
+    HAS_STORM = False
+    STORM_IMP_ERR = traceback.format_exc()
+
+from ansible.module_utils.basic import AnsibleModule, missing_required_lib
+from ansible.module_utils._text import to_native
+
+
+class SSHConfig():
+    def __init__(self, module):
+        self.module = module
+        self.params = module.params
+        self.user = self.params.get('user')
+        self.group = self.params.get('group') or self.user
+        self.host = self.params.get('host')
+        self.config_file = self.params.get('ssh_config_file')
+        self.identity_file = self.params['identity_file']
+        self.check_ssh_config_path()
+        try:
+            self.config = ConfigParser(self.config_file)
+        except FileNotFoundError:
+            self.module.fail_json(msg="Failed to find %s" % self.config_file)
+        self.config.load()
+
+    def check_ssh_config_path(self):
+        if self.user:
+            self.config_file = os.path.join(os.path.expanduser('~%s' % self.user, '.ssh', 'config'))
+        elif self.user in (None, '') and self.config_file is None:
```

Comment:
Why not simply:
```suggestion
        elif self.config_file is None:
```
`self.user` not being `''` / `None` is already covered in the first `if` branch, so it can only be one of these two right now.